src/django/api/tests/reference_image.py

```diff
@@ -0,0 +1,88 @@
+from django.urls import reverse
+from django.utils.log import request_logger
+
+from rest_framework import status
+
+from api.models import ReferenceImage
+
+from .base import BoundarySyncAPITestCase
+
+
+class ReferenceImageViewTests(BoundarySyncAPITestCase):
+    def setUp(self):
+        super().setUp()
+        self.test_image = open("api/tests/data/raleigh_sanborn_map.jpg", "rb")
+
+    def tearDown(self):
+        super().tearDown()
+        self.test_image.close()
+
+    def get_create_payload(self):
+        return {
+            "file": self.test_image,
+            "filename": self.test_image.name,
+            "is_visible": True,
+            "opacity": 100,
+            "uploaded_by": self.contributor.pk,
+        }
+
+    def get_update_payload(self):
+        return {
+            "filename": self.test_image.name,
+            "is_visible": False,
+            "distortion": {},
+            "opacity": 100,
+        }
+
+    def test_no_image_list_endpoint(self):
+        url = reverse("upload_image", args=[1])
+        with self.assertLogs(request_logger, "WARNING"):
+            response = self.client.get(url)
+
+        self.assertEqual(response.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    # Contributor actions
+    def test_contributor_upload_and_update_image(self):
+        url = reverse("upload_image", args=[2])
+        response = self.client.post(url, self.get_create_payload())
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(ReferenceImage.objects.filter(boundary__pk=2).count(), 1)
+
+        update_url = reverse("update_image", args=[2, 1])
+        # Attempt to alter the uploaded_by field, which is illegitimate.
+        updated_image_illegitimate = {
+            **self.get_update_payload(),
+            "uploaded_by": self.administrator.pk,
+        }
+
+        response = self.client.put(
+            update_url, updated_image_illegitimate, format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```

Comment:
Double checking we don't want a `400 Bad Request` here

Reply:
Read Only fields are ignored, as stated in the documentation: https://www.django-rest-framework.org/api-guide/fields/#read_only, so the presence of the read only key does not result in a 400. This design decision is debatable, but in the interest of efficiency I'm going to leave it as is.